input-output-hk/daedalus
# [DDW-1091] Fix storing duplicated "Newsfeed" ID's when marked as read

This PR introduces a fix for duplicated newsfeed IDs when news marked as read.

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1574329069100000)
- [ ] Click multiple times on same newsfeed item and check if "news ID" stored just once in local storage (Follow NOTE steps to reproduce this)

_NOTES_
1. Run app and type in console `daedalus.api.localStorage.unsetReadNews()` to reset all stored news ID's you previously marked as read
2. Reload APP
3. Now you should see that all news are unread. Click on one first time to mark as read and again multiple times on the same to simulate multiple tries (multiple saves to LC).
4. Run command `daedalus.api.localStorage.getReadNews()` to check how much ID's you save to local storage. There shouldn't be duplicated IDs in output like this one:
<img width="434" alt="Screen Shot 2019-11-21 at 10 17 58" src="https://user-images.githubusercontent.com/18653950/69325396-60d66d00-0c4a-11ea-9e65-72338bf92253.png">

* _Especially try to run app outside ITN to see alerts overlay_ *


---

## Review Checklist

### Basics


### Code Quality

### Testing

### After Review
